src/main/java/core/basesyntax/impl/TransactionServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.impl;
+
+import static core.basesyntax.db.FruitStorage.storage;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.TransactionService;
+import java.util.List;
+
+public class TransactionServiceImpl implements TransactionService {
+    private final OperationStrategy operationStrategy;
+
+    public TransactionServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void addToStorage(List<FruitTransaction> activities) {
+        for (FruitTransaction activity : activities) {
+            if (storage.containsKey(activity.getFruit())) {

```

Comment:
u need dao package with dao implementation pattern (dao adds your objects to storage - it is no service level process)
just separate your logic here